internal/receiver/smartagentreceiver/output.go

```diff
@@ -91,11 +91,7 @@ func getMetadataExporters(
 
 	metadataExporters, noClientsSpecified := getDimensionClientsFromMetricsExporters(cfg.DimensionClients, host, nextMetricsConsumer, logger)
 	for _, client := range metadataExporters {
-		if metadataExporter, ok := (*client).(metadata.MetadataExporter); ok {
-			exporters = append(exporters, metadataExporter)
-		} else {
-			logger.Info("cannot send dimension updates to dimension client", zap.Any("client", *client))
-		}
```

Comment:
Why removed?

Reply:
Because as you can see I can append without a cast (means the variable was already that type) :))

Reply:
I think I have a different linter (newer version on my machine) which found this issue.

Reply:
Ah I figured it was returning `[]interface{}` and then checking types but you're right `getDimensionClientsFromMetricsExporters` is already returning `[]metadata.MetadataExporter`